packages/tupaia-web-server/src/routes/EntitySearchRoute.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+
+const DEFAULT_FIELDS = ['code', 'name'];
+
+export type EntitySearchRequest = Request<any, any, any, any>;
+export class EntitySearchRoute extends Route<EntitySearchRequest> {
+  public async buildResponse() {
+    const { query, params, ctx } = this.req;
+    const { projectCode } = params;
+    const { searchString, page = 0, pageSize = 5, fields = DEFAULT_FIELDS } = query;
+
+    const project = (
+      await ctx.services.central.fetchResources('projects', {
+        filter: { code: projectCode },
+        columns: ['entity_hierarchy.name'],
+      })
+    )[0];
+    const entityHierarchyName = project['entity_hierarchy.name'];
+
+    return ctx.services.entity.entitySearch(entityHierarchyName, searchString, {
```

Comment:
```suggestion
    return ctx.services.entity.entitySearch(projectCode, searchString, {
```
You're all good using the project code here. For some time now we've committed to there being a one to one mapping between projects and entity hierarchies, and the project code and the hierarchy name are always the same. It's an assumption made in various parts of the code.

Reply:
Oh cool. I run this same code against a few routes so I'll rip them all out. 👍 Big fan of deleting code